### PR TITLE
Clean up dependencies

### LINKED ISSUE #1024: Cargo Udeps
# Unused dependencies 
| Crate | Package | Type |
|----:|-------:|-------:|
|namada_apps 0.35.1|assert_matches|normal|
|namada_apps 0.35.1|pretty_assertions|normal|
|namada_apps 0.35.1|signal-hook|normal|
|namada_apps 0.35.1|tonic|normal|
|namada_apps 0.35.1|winapi|normal|
|namada_apps 0.35.1|assert_matches|development|
|namada_core 0.35.1|rayon|normal|
|namada_core 0.35.1|rayon|development|
|namada_ethereum_bridge 0.35.1|tendermint-rpc|normal|
|namada_macros 0.35.1|lazy_static|normal|
|namada_macros 0.35.1|paste|normal|
|namada_migrations 0.35.1|borsh|normal|
|namada_migrations 0.35.1|data-encoding|normal|
|namada_migrations 0.35.1|serde|normal|
|namada_proof_of_stake 0.35.1|derivative|normal|
|namada_proof_of_stake 0.35.1|derivative|development|
|namada_state 0.35.1|namada_trans_token|normal|
|namada_storage 0.35.1|namada_gas|normal|
|namada_storage 0.35.1|namada_tx|normal|
|namada_tests 0.35.1|clap|normal|
|namada_tests 0.35.1|copy_dir|normal|
|namada_tests 0.35.1|tendermint-light-client|normal|

### PR DESCRIPTION
**Pull request by brentstone**
## Describe your changes
Closes #1024.
Mostly addresses #1023.

## Indicate on which release or other PRs this topic is based on
v0.35.1

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

File: crates/apps/Cargo.toml [hunk 1]
```diff
@@ -146,23 +143,22 @@ tendermint-rpc.workspace = true
...
 tokio = {workspace = true, features = ["full"]}
 toml.workspace = true
-tonic.workspace = true
 tower-abci.workspace = true
 tower.workspace = true
 tracing-appender.workspace = true
 tracing-log.workspace = true
 tracing-subscriber = { workspace = true, features = ["std", "json", "ansi", "tracing-log"]}
 tracing.workspace = true
-winapi.workspace = true
```
Comment (tzemanovic) [inline crates/apps/Cargo.toml:156]: this is used with conditional compilation - we should put in `[target.'cfg(windows)'.dependencies]` instead

File: crates/core/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ version.workspace = true
 [features]
 default = []
 mainnet = []
-wasm-runtime = ["rayon"]
+wasm-runtime = []
```
Comment (tzemanovic) [inline crates/core/Cargo.toml:18]: this features is never used in `crates/core` let's just remove it altogether

File: crates/sdk/Cargo.toml [hunk 3]
```diff
@@ -97,6 +97,7 @@ borsh-ext.workspace = true
 circular-queue.workspace = true
 data-encoding.workspace = true
 derivation-path.workspace = true
+duration-str = "0.10.0"
```
Comment (tzemanovic) [inline crates/sdk/Cargo.toml:100]: let's declare this version in the workspace `Cargo.toml` - I think only exception to this should be the `crates/apps` for binaries specific deps

Comment (tzemanovic) [reply crates/sdk/Cargo.toml:100]: there are couple more in here too

**Review by tzemanovic — CHANGES_REQUESTED** [review]
couple small things - other than that LGTM!

File: crates/sdk/Cargo.toml (hunk 3 above)
Comment (brentstone) [reply crates/sdk/Cargo.toml]: fixed in a8c3d66b9b557ed4e73156615632905a247a3c04

File: crates/core/Cargo.toml (hunk 2 above)
Comment (brentstone) [reply crates/core/Cargo.toml]: fixed in a8c3d66b9b557ed4e73156615632905a247a3c04

File: crates/apps/Cargo.toml (hunk 1 above)
Comment (brentstone) [reply crates/apps/Cargo.toml:156]: fixed in a8c3d66b9b557ed4e73156615632905a247a3c04

**Review by tzemanovic — APPROVED** [review]

**Pull request merged**